src/client_config.rs

```diff
@@ -0,0 +1,192 @@
+                }
+                "-d" | "--download" => {
+                    config.mode = Mode::Download;
+                }
+                "-f" | "--file" => {
+                    if let Some(file_str) = args.next() {
+                        config.file = file_str;
+                    } else {
+                        return Err("Missing send directory after flag".into());
+                    }
+                }
+                "-h" | "--help" => {
+                    println!("TFTP Client\n");
+                    println!("Usage: tftpd client [OPTIONS] <-f | --file> <File>\n");
+                    println!("Options:");
+                    println!("  -i, --ip-address <IP ADDRESS>\tIp address of the server (default: 127.0.0.1)");
+                    println!("  -p, --port <PORT>\t\tPort of the server (default: 69)");
+                    println!("  -b, --blocksize <number>\tSets the blocksize (default: 512)");
+                    println!("  -w, --windowsize <number>\tSets the windowsize (default: 1)");
+                    println!("  -t, --timeout <seconds>\tSets the timeout in seconds (default: 5)");
+                    println!("  -u, --upload\t\t\tSets the client to upload mode, Ignores all previous download flags");
+                    println!("  -d, --download\t\tSet the client to download mode, Invalidates all previous upload flags");
+                    println!(
+                        "  -f, --file\t\t\tLocal file to upload if in upload mode. Name of the remote file if in download mode"
+                    );
+                    println!("  -sd, --save-directory <DIRECTORY>\tSet the directory to save files when in Download Mode (default: the directory setting)");
+                    println!("  -h, --help\t\t\tPrint help information");
+                    process::exit(0);
+                }
+
+                invalid => return Err(format!("Invalid flag: {invalid}").into()),
+            }
+        }
+
+        Ok(config)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use ClientConfig;
+
+    use super::*;
+
+    #[test]
+    fn parses_full_config() {
+        let config = ClientConfig::new(
+            [
+                "client",
+                "-i",
+                "0.0.0.0",
+                "-p",
+                "1234",
+                "-f",
+                "test.yaml",
+                "-sd",
+                "/",
+                "-d",
+                "-u",
+                "-b",
+                "1024",
+                "-w",
+                "2",
+            ]
+            .iter()
+            .map(|s| s.to_string()),
+        )
+        .unwrap();
+
+        assert_eq!(config.remote_ip_address, Ipv4Addr::new(0, 0, 0, 0));
+        assert_eq!(config.port, 1234);
+        assert_eq!(config.save_directory, PathBuf::from("/"));
+        assert_eq!(config.windowsize, 2);
+        assert_eq!(config.blocksize, 1024);
+        assert_eq!(config.mode, Mode::Upload);
+    }
```

Comment:
Can we also add a test for partial parsing here? Not required, but would be great for future contributors.